app/Models/Device.php

```diff
@@ -141,6 +141,19 @@ public static function findByIp($ip)
 
         return null;
     }
+    
+    public static function findByHostnameOrIp($hostnameOrIp)
+    {
+        $device = Device::findByHostname($hostnameOrIp);
```

Comment:
Instead of doing two sql queries, you can just check to see if the input is an IP address.
Also, on the second query, you can just return it, you don't have to check if it is null to see if you should return it or null.

`\LibreNMS\Util\IP::isValid($hostnameOrIp)`

Reply:
> Instead of doing two sql queries, you can just check to see if the input is an IP address.

Do you mean like this to only "findByIp" if parameter is valid ip?
```
if (IP::isValid($hostnameOrIp)) {
    $device = Device::findByIp($hostnameOrIp);
    if (isset($device)) {
        return $device;
    }
}
return Device::findByHostname($hostnameOrIp);
```

> Also, on the second query, you can just return it, you don't have to check if it is null to see if you should return it or null.

Yes you are right. Will change this